implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/listener/ranch.ex

```diff
@@ -0,0 +1,72 @@
+if Code.ensure_loaded?(:ranch) do
+  defmodule Ockam.Transport.TCP.Listener.Ranch do
+    @moduledoc false
+
+    use Ockam.Worker
+
+    @doc false
+    @impl true
+    def setup(options, state) do
+      ip = Keyword.get_lazy(options, :ip, &default_ip/0)
+      state = Map.put(state, :ip, ip)
+
+      port = Keyword.get_lazy(options, :port, &default_port/0)
+      state = Map.put(state, :port, port)
+
+      ref = make_ref()
+      transport = :ranch_tcp
+      transport_options = [port: port]
+      protocol = __MODULE__.Protocol
+      protocol_options = []
+
+      with {:ok, _apps} <- Application.ensure_all_started(:ranch),
+           :ok <- start_listerner(ref, transport, transport_options, protocol, protocol_options) do
```

Comment:
assuming the Telemetry is wrapped in this call, however I think this is another typo: `start_listerner` -> `start_listener`

Reply:
fixed